src/compiler/factory.ts

```diff
@@ -1065,7 +1065,9 @@ namespace ts {
     }
 
     export function updatePropertyAccess(node: PropertyAccessExpression, expression: Expression, name: Identifier) {
-        Debug.assert(!(node.flags & NodeFlags.OptionalChain), "Cannot update a PropertyAccessChain using updatePropertyAccess. Use updatePropertyAccessChain instead.");
+        if (isOptionalChain(node)) {
+            return updatePropertyAccessChain(node, expression, node.questionDotToken, name);

```

Comment:
I imagine if we had a way to emit a deprecation warning, we'd do it here.

Reply:
I'm working on that elsewhere...